akka-actor/src/main/scala/akka/actor/FaultHandling.scala

```diff
@@ -155,6 +155,7 @@ object SupervisorStrategy extends SupervisorStrategyLowPriorityImplicits {
     case _: ActorInitializationException ⇒ Stop
     case _: ActorKilledException         ⇒ Stop
     case _: DeathPactException           ⇒ Stop
+    case m: IncompatibleClassChangeError ⇒ detectAkkaBinaryIncompatibilityWarningAndStop(m)
```

Comment:
It's not about when it happens, it is tha NonFatal will not catch it and therefore it will not end up in supervisor strategy. Unless I'm wrong :)